### PR TITLE
Canvas: Chore clean up betterer

### PR DESCRIPTION
**Pull request by nmarrs**
Clean up canvas betterer complaints. For a good amount waived away betterer warnings for battle tested code but fixed / addressed some issues.

Main reason behind these changes is to reduce risk of future cleanup refactors from breaking critical functionality which occurred recently with https://github.com/grafana/grafana/pull/87363

Did not address all canvas betterer issues, the last two files will require a bit more extensive refactor that I will follow-up with to get rid of canvas's [barrel files](https://uglow.medium.com/burn-the-barrel-c282578f21b6)

### CODE REVIEW

File: public/app/features/canvas/runtime/scene.tsx [hunk 1]
```diff
@@ -838,12 +840,9 @@ export class Scene {
   }
 }
 
-const getStyles = stylesFactory((theme: GrafanaTheme2) => ({
-  wrap: css`
-    overflow: hidden;
-    position: relative;
-  `,
-  selected: css`
-    z-index: 999 !important;
```
Comment (nmarrs) [inline public/app/features/canvas/runtime/scene.tsx:847]: this is the riskiest change here, did some digging and it appears that this value was never actually being used! The class was being applied to movable's control container that has an overriding z-index of 3000 (vs the 999 being set previously)

[See docs](https://daybrush.com/moveable/release/latest/doc/Moveable.html#className) and pic below
zindex seems useless

**Comment by baldm0mma** [comment]
Nice! Maybe I don't understand, but what is the benefit of adding inline linter disabling comments instead of having the betterer warnings? Genuinely curious. :)

**Comment by nmarrs** [comment]
> Nice! Maybe I don't understand, but what is the benefit of adding inline linter disabling comments instead of having the betterer warnings? Genuinely curious. :)

Stuff in betterer indicates a code smell / something that should be addressed / cleaned up - basically goal of betterer is to flag and indicate code smells / quality issues in codebase and overall goal is for betterer results file to become empty over time. In some cases these code smells (like avoiding using the `any` type) make sense and so adding the inline linter disable basically codifies the code smell as "acceptable" and removes it from the watchful eyes of betterer.

Greater context around reason for these changes right now is that the betterer file can be a source of low hanging refactors - [one of these recent refactors ended up breaking core functionality in canvas](https://github.com/grafana/grafana/pull/87363) 😬 so by cleaning up / managing our betterer issues ourselves we can avoid this happening in the future / own our code's quality (and destiny 😅)

**Review by baldm0mma — APPROVED** [review]
Thanks for the explanation! Great work!

**Pull request merged**